File: src/main/scala/org/excavator/boot/optaplanner/domain/Lesson.scala
package org.excavator.boot.optaplanner.domain

import org.optaplanner.core.api.domain.entity.PlanningEntity
import org.optaplanner.core.api.domain.variable.PlanningVariable

import scala.beans.BeanProperty

@PlanningEntity
case class Lesson(@BeanProperty id:Long,
            @BeanProperty subject:String,
            @BeanProperty teacher:String,
            @BeanProperty studentGroup:String){

  @PlanningVariable(valueRangeProviderRefs = Array("roomRange"))
  @BeanProperty
  var timeslot:Timeslot = null

  @PlanningVariable(valueRangeProviderRefs = Array("roomRange"))
  @BeanProperty
  var room:Room = null
}

File: src/main/java/org/excavator/boot/optaplanner/domain/Lesson.java
package org.excavator.boot.optaplanner.domain;

import org.optaplanner.core.api.domain.variable.PlanningVariable;

public class Lesson {
    private long id;

    private String subject;

    private String teacher;

    private String studentGroup;

    @PlanningVariable(valueRangeProviderRefs = "timeslotRange")
    private Timeslot timeslot;

    @PlanningVariable(valueRangeProviderRefs = "roomRange")
    private Room room;

    private Lesson() {
    }

    public Lesson(long id, String subject, String teacher, String studentGroup, Timeslot timeslot, Room room) {
        this.id = id;
        this.subject = subject;
        this.teacher = teacher;
        this.studentGroup = studentGroup;
        this.timeslot = timeslot;
        this.room = room;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public String getTeacher() {
        return teacher;
    }

    public void setTeacher(String teacher) {
        this.teacher = teacher;
    }

    public String getStudentGroup() {
        return studentGroup;
    }

    public void setStudentGroup(String studentGroup) {
        this.studentGroup = studentGroup;
    }

    public Timeslot getTimeslot() {
        return timeslot;
    }

    public void setTimeslot(Timeslot timeslot) {
        this.timeslot = timeslot;
    }

    public Room getRoom() {
        return room;
    }

    public void setRoom(Room room) {
        this.room = room;
    }
}
